src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -137,73 +114,83 @@ export const SEQUENTIAL_COLOR_SCHEMES = ['blues', 'greens', 'greys', 'purples',
-    } else if (nestedProp === 'title') {
-      return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-    } else if (nestedProp === 'type') {
-      const propertySchema = {
-        ...baseUISchema,
-        enum: (vlSchema as any).definitions.Legend.properties.type.enum,
-      };
-      return generateSchema(prop, nestedProp, DEFAULT_SELECT_UISCHEMA, propTab, propertySchema);
-    }
-  } else if (prop === 'format') {
-    return generateSchema(prop, nestedProp, DEFAULT_TEXT_UISCHEMA, propTab, baseUISchema);
-  } else {
-    throw new Error('Property combination not recognized');
+      return generateScaleEditorSchema(nestedProp, scaleTypes, fieldDef, baseUISchema);
+    case 'axis':
+      return generateAxisEditorSchema(nestedProp, channel as Channel, baseUISchema, propertyKey);
+    case 'stack':
+      return generateStackEditorSchema(baseUISchema);
+    case 'legend':
+      return generateLegendEditorSchema(nestedProp, baseUISchema, propertyKey);
+    case 'format':
+      return generateFormatEditorSchema(baseUISchema);
+    default:
+      throw new Error('Property combination not recognized');
+  }
+}
+
+function generateFormatEditorSchema(baseUISchema: SchemaProperty) {
+  return generateTextBoxSchema('format', '', baseUISchema, 'string');
+}
+
+function generateLegendEditorSchema(nestedProp: string, baseUISchema: SchemaProperty, propertyKey: string) {
+  switch (nestedProp) {
+    case 'orient':
+      const orientVals = (vlSchema as any).definitions.LegendOrient.enum;
+      return generateSelectSchema(propertyKey, orientVals, baseUISchema);
+    case 'title':
+      return generateTextBoxSchema(propertyKey, '', baseUISchema, 'string');
+    case 'type':
+      const typeVals = (vlSchema as any).definitions.Legend.properties.type.enum;
+      return generateSelectSchema(propertyKey, typeVals, baseUISchema);
+    default:
+      throw new Error('Property combination not recognized');
+  }
+}
+
+function generateStackEditorSchema(baseUISchema: SchemaProperty) {
+  const enumVals = (vlSchema as any).definitions.StackOffset.enum;
+  return generateSelectSchema('stack', enumVals, baseUISchema);
+}
+
+function generateAxisEditorSchema(nestedProp: string, channel: Channel, baseUISchema: SchemaProperty,
+                                  propertyKey: string) {
+  switch (nestedProp) {
+    case 'orient':
+      const enumVals = channel === 'y' ? ['left', 'right'] : ['top', 'bottom'];
+      return generateSelectSchema(propertyKey, enumVals, baseUISchema);
+    case 'title':
+      return generateTextBoxSchema(propertyKey, '', baseUISchema, 'string');
+    default:
+      throw new Error('Property combination not recognized');
   }
 }
 
+function generateScaleEditorSchema(nestedProp: string, scaleTypes: string[], fieldDef: ShelfFieldDef,
+                                   baseUISchema: SchemaProperty) {
+  const propertyKey = nestedProp ? 'scale_' + nestedProp : 'scale';
```

Comment:
1) when would there be no nestedProp for scale?

2) Also, why do some methods have propertyKey in the signature, but some do not? Please be consistent. 

If you haven't, always look at all of your diff via Source Tree or equivalent tools before committing.  This will improve your code quality. :)